truewiki/views/edit.py

```diff
@@ -125,4 +127,9 @@ def view(user, page: str) -> web.Response:
         )
 
     body = source.create_body(wiki_page, user, "Edit", new_page=page)
-    return web.Response(body=body, content_type="text/html")
+
+    response = web.Response(body=body, content_type="text/html")
+    response.set_cookie(
+        EDIT_TIME_COOKIE_NAME, singleton.STORAGE.get_file_timestamp(wiki_page.page_ondisk_name(page)), httponly=True
+    )
```

Comment:
I do not really understand why you went for a cookie for this? That feels like the wrong way to solve it?

Isn't it better to add it in the `form`, like a sort of CSRF token? (just one that gives a polite error back if it fails :P)

Reply:
You're right, that would be better, but I honestly don't know how to go about doing that. I wanted to send the timestamp back in the POST request but it seems like all the payload data is based on what the page inputs are, and I don't want the timestamp to be user-editable.

Reply:
Normally you use an `input type="hidden"`  for this. And yes, a clever user can change it. But what does he win by doing that? It is not something to really worry about :)